test/unit/parser_test.rb

```diff
@@ -786,13 +795,34 @@ def test_cast_with_field
     assert_equal 'cast', expression[:field_function]
     assert_equal "'100000'", expression[:condition]
     assert_equal(:character, expression[:field_function_type])
+
+    assert_equal 'cast', expression[:field_manipulations][:function_name]
+    assert_equal :function, expression[:field_manipulations][:type]
+    assert_equal ['ListPrice', 'character'], expression[:field_manipulations][:args].map {|v| v[:value]}
   end
 
   def test_cast_with_invalid_type
     parser_errors("cast(ListPrice, 'bogus') Eq '10'")
     parser_errors("ListPrice Eq cast('10', 'bogus')")
   end
 
+  test 'nested functions on field side' do
+    @parser = Parser.new
+    filter = "tolower(toupper(City)) Eq 'Fargo'"
```

Comment:
I think this could use a test covering a case with more than one parameter when there's function testing:

```
concat(tolower(City), 'b') Eq 'fargob'
```

Reply:
Good call 👍